### ISSUE TITLE
Why `box_loss / labels.numel()` instead of `box_loss / labels_pos.numel()`?

### ISSUE THREAD

**Issue by chenzhutian** [facebookresearch/maskrcnn-benchmark#486]
## ❓ Questions and Help
For the box_loss here:
https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/maskrcnn_benchmark/modeling/roi_heads/box_head/loss.py#L165

According to my understanding, the box_loss is only defined on postive samples.
So why not `box_loss = box_loss / labels_pos.numel()` ?

**Comment by fmassa** [comment]
This is to maintain consistency with the implementation in Detectron, which used their custom SmoothL1Loss with `inside_weights` and `outside_weights`.

They divide the loss by `labels.numel()`, see [here](https://github.com/pytorch/pytorch/blob/cca247635c6edb323176eeac7a18d3e9ab71c558/modules/detectron/smooth_l1_loss_op.cc#L41). We could probably change it to `labels_pos.numel()`, but it would involve changing the scaling of the losses.

For completeness, here is how it is implemented in Detectron: [the inside_weights and outside_weights](https://github.com/facebookresearch/Detectron/blob/8a4723835cb163c3c02d90593f2f4f4ef1f45f4b/detectron/roi_data/fast_rcnn.py#L173-L178) and the [smooth l1 loss computation](https://github.com/facebookresearch/Detectron/blob/8a4723835cb163c3c02d90593f2f4f4ef1f45f4b/detectron/modeling/fast_rcnn_heads.py#L81-L88)
